miio/g1vacuum.py

```diff
@@ -0,0 +1,362 @@
+            {'did': 'main_brush_time_left', 'siid': 14, 'piid': 2, 'code': 0, 'value': 17959},
+            {'did': 'side_brush_life_level', 'siid': 15, 'piid': 1}, 
+            {'did': 'side_brush_time_left', 'siid': 15, 'piid': 2}, 
+            {'did': 'filter_life_level', 'siid': 11, 'piid': 1}, 
+            {'did': 'filter_time_left', 'siid': 11, 'piid': 2},
+            {'did': 'clean_area', 'siid': 9, 'piid': 1}, 
+            {'did': 'clean_time', 'siid': 9, 'piid': 2}
+            ]
+
+    """
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def charge_state(self) -> G1ChargeState:
+        """Charging State."""
+        return G1ChargeState[G1ChargeState(self.data["charge_state"]).name]
+
+    @property
+    def error_code(self) -> int:
+        """Error code as returned by the device."""
+        return int(self.data["error_code"])
+
+    @property
+    def error(self) -> str:
+        """Human readable error description, see also :func:`error_code`."""
+        try:
+            return error_codes[self.error_code]
+        except KeyError:
+            return "Definition missing for error %s" % self.error_code
+
+    @property
+    def state(self) -> G1State:
+        """Vacuum Status."""
+        return G1State(self.data["state"])
+
+    @property
+    def fan_speed(self) -> G1FanSpeed:
+        """Fan Speed."""
+        return G1FanSpeed(self.data["fan_speed"])
+
+    @property
+    def operating_mode(self) -> G1VacuumMode:
+        """Operating Mode."""
+        return G1VacuumMode(self.data["operating_mode"])
+
+    @property
+    def mop_state(self) -> G1MopState:
+        """Mop State."""
+        return G1MopState(self.data["mop_state"])
+
+    @property
+    def water_level(self) -> G1WaterLevel:
+        """Water Level."""
+        return G1WaterLevel(self.data["water_level"])
+    
+    @property
+    def main_brush_life_level(self) -> int:
+        """Main Brush Life Level in %."""
+        return self.data["main_brush_life_level"]
+
+    @property
+    def main_brush_time_left(self) -> timedelta:
+        """Main Brush Remaining Time in Minutes."""
+        return timedelta(seconds=self.data["main_brush_time_left"])
+  
+    @property
+    def side_brush_life_level(self) -> int:
+        """Side Brush Life Level in %."""
+        return self.data["side_brush_life_level"]
+
+    @property
+    def side_brush_time_left(self) -> timedelta:
+        """Side Brush Remaining Time in Minutes."""
+        return timedelta(minutes=self.data["side_brush_time_left"])
+
+    @property
+    def filter_life_level(self) -> int:
+        """Filter Life Level in %."""
+        return self.data["filter_life_level"]
+
+    @property
+    def filter_time_left(self) -> timedelta:
+        """Filter Remaining Time in Minutes."""
```

Comment:
```suggestion
        """Filter remaining time."""
```
No need to mention the unit when using timedleta, please also avoid Capital Casing in docstrings. This applies to docstrings of other properties & methods too, but I'm not going to separately mark them so please do a quick check on those.